test/functional/feature_assumeutxo.py

```diff
@@ -0,0 +1,246 @@
+
+        self.log.info("-- Testing assumeutxo + some indexes + pruning")
+
+        assert_equal(n0.getblockcount(), SNAPSHOT_BASE_HEIGHT)
+        assert_equal(n1.getblockcount(), START_HEIGHT)
+
+        self.log.info(f"Creating a UTXO snapshot at height {SNAPSHOT_BASE_HEIGHT}")
+        dump_output = n0.dumptxoutset('utxos.dat')
+
+        assert_equal(
+            dump_output['txoutset_hash'],
+            'ef45ccdca5898b6c2145e4581d2b88c56564dd389e4bd75a1aaf6961d3edd3c0')
+        assert_equal(dump_output['nchaintx'], 300)
+        assert_equal(n0.getblockchaininfo()["blocks"], SNAPSHOT_BASE_HEIGHT)
+
+        # Mine more blocks on top of the snapshot that n1 hasn't yet seen. This
+        # will allow us to test n1's sync-to-tip on top of a snapshot.
+        self.generate(n0, nblocks=100, sync_fun=no_sync)
+
+        assert_equal(n0.getblockcount(), FINAL_HEIGHT)
+        assert_equal(n1.getblockcount(), START_HEIGHT)
+
+        assert_equal(n0.getblockchaininfo()["blocks"], FINAL_HEIGHT)
+
+        self.log.info(f"Loading snapshot into second node from {dump_output['path']}")
+        loaded = n1.loadtxoutset(dump_output['path'])
+        assert_equal(loaded['coins_loaded'], SNAPSHOT_BASE_HEIGHT)
+        assert_equal(loaded['base_height'], SNAPSHOT_BASE_HEIGHT)
+
+        monitor = n1.getchainstates()
+        assert_equal(monitor['normal']['blocks'], START_HEIGHT)
+        assert_equal(monitor['snapshot']['blocks'], SNAPSHOT_BASE_HEIGHT)
+        assert_equal(monitor['snapshot']['snapshot_blockhash'], dump_output['base_hash'])
+
+        assert_equal(n1.getblockchaininfo()["blocks"], SNAPSHOT_BASE_HEIGHT)
+
+        PAUSE_HEIGHT = FINAL_HEIGHT - 40
+
+        self.log.info("Restarting node to stop at height %d", PAUSE_HEIGHT)
+        self.restart_node(1, extra_args=[
+            f"-stopatheight={PAUSE_HEIGHT}", *self.extra_args[1]])
+
+        # Finally connect the nodes and let them sync.
+        self.connect_nodes(0, 1)
+
+        n1.wait_until_stopped(timeout=5)
+
+        self.log.info("Checking that blocks are segmented on disk")
+        assert self.has_blockfile(n1, "00000"), "normal blockfile missing"
+        assert self.has_blockfile(n1, "00001"), "assumed blockfile missing"
+        assert not self.has_blockfile(n1, "00002"), "too many blockfiles"
+
+        self.log.info("Restarted node before snapshot validation completed, reloading...")
+        self.restart_node(1, extra_args=self.extra_args[1])
+        self.connect_nodes(0, 1)
+
+        self.log.info(f"Ensuring snapshot chain syncs to tip. ({FINAL_HEIGHT})")
+        wait_until_helper(lambda: n1.getchainstates()['snapshot']['blocks'] == FINAL_HEIGHT)
+        self.sync_blocks(nodes=(n0, n1))
+
+        self.log.info("Ensuring background validation completes")
+        # N.B.: the `snapshot` key disappears once the background validation is complete.
+        wait_until_helper(lambda: not n1.getchainstates().get('snapshot'))
```

Comment:
In 42cae39356fd20d521aaf99aff1ed85856f3c9f3 "test: add feature_assumeutxo functional test"

These `wait_until_helper`s and `self.wait_until` can timeout if the machine is kinda slow, or if running in valgrind.

Reply:
This have been related to #28589, if not #28586 can be reopened to address this